docs/Developers/Debug_Bears.rst

```diff
@@ -0,0 +1,52 @@
+Debug Bears
+===========
+
+This document provide a brief overview of coala's Debugging Interface.
+Debugging Interface will help user to debug the Bear code and move through
+it using ``pdb`` Interface
+
+There are two way to invoke the coala's Debgger
+
+Command Line Interface
+----------------------
+
+User can specify the bear they want to debug using ``--debug-bears`` i.e.,
+
+.. code:: shell
+
+    $ coala -b PEP8Bear,MypyBear -f <filename> --debug-bears PEP8Bear
+
+If bear name is not specified in ``--debug-bears`` argument then it will by
+default debug all the bears passed through ``--bears`` or ``-b`` argument
+**i.e.,**
+
+.. code:: shell
+
+    $ coala -b PEP8Bear,MypyBear -f <filename> --debug-bears
+
+Configuration Files - coafiles
+------------------------------
+
+User can debug the bears using ``.coafile`` **i.e.,**
+
+::
+
+    [<section name>]
+    bears = PEP8Bear,MypyBear
+    files = <filename>
+    debug_bears = PEP8Bear
+
+**or**
+
+::
+
+    [<section name>]
+    bears = PEP8Bear,MypyBear
+    files = <filename>
+    debug_bears = True
+
+.. note::
+
+    ``Pdb`` ``do_q`` (quit) has been remapped so that coala can continue its
+    normal execution without any error. So, ``quit`` or `q` will first clear
+    all the breakspoints then continue ( ``c`` or ``continue`` ) execution.
```

Comment:
Hm... Not sure that's too much implementation detail. But let's keep it for now, I'll revise it again when the other stuff is in place ;)